AspNetCore3ChildlifetimeScope/Program.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Threading.Tasks;
+using AspNetCore3ChildlifetimeScope.Dependencies;
+using Autofac;
+using Autofac.Extensions.DependencyInjection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+
+namespace AspNetCore3ChildlifetimeScope
+{
+    public static class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            var containerBuilder = new ContainerBuilder();
+
+            containerBuilder.RegisterType<SingletonRootDependency>()
+                .As<ISingletonRootDependency>()
+                .AsSelf()
+                .SingleInstance();
+
+            var container = containerBuilder.Build();
+
+            var scope = container.BeginLifetimeScope("RootChildScope", builder => builder.RegisterType<TransientRootChildScopeDependency>().As<ITransientRootChildScopeDependency>().AsSelf().InstancePerDependency());
+
+            using var hostApplicationA = CreateHostForApplicationA(scope, args);
+
+            using var hostApplicationB = CreateHostForApplicationB(scope, args);
+
+            await Task.WhenAll(hostApplicationA.RunAsync(), hostApplicationB.RunAsync());
+        }
+
+        private static IHost CreateHostForApplicationA(ILifetimeScope container, string[] args)
```

Comment:
Call the `ILifetimeScope` parameter `scope` or something similar to show that it doesn't need a container. If you're skimming the code, it's easy to see a `container` getting passed in and not realize that it's really not an `IContainer`.